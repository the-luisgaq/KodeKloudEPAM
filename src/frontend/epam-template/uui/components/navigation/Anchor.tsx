import { Anchor as uuiAnchor, AnchorProps } from '@epam/uui-components';
import { withMods } from '@epam/uui-core';
import css from './Anchor.module.scss';

export const Anchor = withMods<AnchorProps, AnchorProps>(uuiAnchor, () => [css.root]);
