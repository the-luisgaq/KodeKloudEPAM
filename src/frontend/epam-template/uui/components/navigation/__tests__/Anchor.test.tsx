import React from 'react';
import { renderSnapshotWithContextAsync } from '@epam/uui-test-utils';
import { Anchor } from '../Anchor';

describe('Anchor', () => {
    it('should rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(<Anchor href="#">Test Anchor</Anchor>);
        expect(tree).toMatchSnapshot();
    });

    it('should rendered with props correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(
            <Anchor href="https://uui.epam.com" isDisabled={ true }>
                Test Anchor
            </Anchor>,
        );
        expect(tree).toMatchSnapshot();
    });
});
