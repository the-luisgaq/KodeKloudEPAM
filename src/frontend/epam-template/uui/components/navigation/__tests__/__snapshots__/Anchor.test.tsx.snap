// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor should rendered correctly 1`] = `
<DocumentFragment>
  <a
    class="uui-button-box uui-enabled -clickable uui-anchor container root"
    href="#"
    tabindex="0"
  >
    Test Anchor
  </a>
</DocumentFragment>
`;

exports[`Anchor should rendered with props correctly 1`] = `
<DocumentFragment>
  <a
    aria-disabled="true"
    class="uui-button-box uui-disabled -clickable uui-anchor container root"
    tabindex="-1"
  >
    Test Anchor
  </a>
</DocumentFragment>
`;
